File: code/backend/python/app/core/services/echo_service.py
import sys
import time
from app.core.ports.echo_port import EchoPort


class EchoService(EchoPort):
    def __init__(self, server_status: str, server_properties: dict[str, str]):
        self.server_status = server_status
        self.server_properties = server_properties

    def get_server_status(self) -> str:
        return self.server_status

    def get_timestamp(self) -> int:
        return int(time.time() * 1000)

    def get_server_properties(self) -> dict[str, str]:
        props = dict(self.server_properties)
        props["pythonVersion"] = f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
        return props

File: code/backend/node/src/core/services/EchoService.ts
export class EchoService {
  getServerStatus(): string {
    return 'OK';
  }

  getTimestamp(): number {
    return Date.now();
  }

  getServerProperties(): Record<string, string> {
    return {
      status: 'OK',
      version: '1.0.0',
      timestamp: new Date().toISOString(),
    };
  }
}
